quickwit-common/src/uri.rs

```diff
@@ -28,6 +29,26 @@ const FILE_PROTOCOL: &str = "file";
 
 const PROTOCOL_SEPARATOR: &str = "://";
 
+#[derive(Debug, PartialEq)]
+pub enum Extension {
```

Comment:
I got carried away trying to refactor and commonize the loading logic of `QuickwitConfig`, `IndexConfig`, and `SourceConfig`. Those are the remains of the aborted refactor. I thought that having a `.extension` method on `Uri` might be helpful in the future, but I would be happy to remove it if judged premature.